src/firefly/js/ui/DateTimePickerField.jsx

```diff
@@ -208,13 +208,19 @@ function handleOnChange(momentVal, params, fireValueChange) {
     fireValueChange({value, valid, message});
 }
 
+function convertMomentToMJD(m) {
+    return m.isValid() ? `${(m.valueOf()/msecPerDay + DiffUnixMJD)}`: '';
+}
+
 export const DateTimePickerField = fieldGroupConnector(DateTimePicker, getProps, propTypes);
 
 const DiffUnixMJD = 40587.0;
 const DiffUnixJD = 2440587.5;
 const DiffMJDJD = 2400000.5;
 const msecPerDay = 86400000.0;
 
+const MIN_VAL_MJD = 1; // for ISO 1858-11-18T00:00:00Z
+const MAX_VAL_MJD = 2973483;  // for ISO 9999-12-31T00:00:00Z
```

Comment:
It is good to have consistent validation range for both ISO & MJD. However, what if the user enters '0' for MJD, or '1858-11-17T00:00:00Z' or ' 9999-12-31T00:01:00Z' for ISO, those number will be validated as invalid. Actually those numbers are not that meaningful for temporal search, however theoretically how to explain that those numbers are 'invalid MJD' or 'invalid ISO'. We may inform the user the valid range in the documentation or the tooltip in UI.

Reply:
The valid date-time range is random, we can always adjust it to something better. We give user the supported range in the error message next to the field ("date is out of range ... to ..."). I think this is good enough for now.